agent/api/task/service_connect.go

```diff
@@ -65,3 +80,55 @@ type DNSConfigEntry struct {
 	HostName string `json:"hostName"`
 	Address  string `json:"address"`
 }
+
+// ExtractSCConfigFromACSAttachment finds the service connect attachment, and extracts the service connect container name and service connect config value.
+func ExtractSCConfigFromACSAttachment(acsTask *ecsacs.Task) (*ServiceConnectConfig, error) {
+	if acsTask.Attachments == nil {
+		return nil, nil
+	}
+	var scAttachment *ecsacs.Attachment
+	containerName := ""
+	for _, attachment := range acsTask.Attachments {
+		if aws.StringValue(attachment.AttachmentType) == serviceConnectAttType {
+			scAttachment = attachment
+			break
+		}
+	}
+	// no service connect attachment is found
+	if scAttachment == nil {
+		return nil, nil
+	}
+	// translate ecsacs.Attachment.Value to ServiceConnectConfig
+	scConfigValue := &ServiceConnectConfig{}
+	for _, property := range scAttachment.AttachmentProperties {
+		switch aws.StringValue(property.Name) {
+		case serviceConnectAttProSCConfigName:
+			s, _ := strconv.Unquote(string(*property.Value))
+			if err := json.Unmarshal([]byte(s), scConfigValue); err != nil {
+				return nil, err
+			}
+		case serviceConnectAttProSCContainerName:
+			containerName = aws.StringValue(property.Value)
+		}
+	}
```

Comment:
nit - maybe add a default case to report error on unrecognized properties

Reply:
```
default:
    return nil, errors.Errorf("received an unrecognized attachment property: %s", property.String())
```
is added in commit 8e69368bc5c1d7bd3ab71391103f4b6a69f06054. Thanks!